src/main/java/com/develhope/spring/order/service/OrderService.java

```diff
@@ -41,4 +42,15 @@ public void deleteOrderById(long id) {
     public void deleteAllOrders() {
         orderRepository.deleteAll();
     }
+
+    public void updateOrderStatus(long orderId) {
+        Order order = orderRepository.findById(orderId).get();
+        if (order.getOrderStatus() == OrderStatus.ORDERED) {
```

Comment:
quello che intendeva Emanuele, almeno credo, era che il metodo del service oltre l'id dell'order per identificarlo dovrebbe prendere in ingresso anche l'orderStatus che si vuole assegnare, questo perchè ad esempio dobbiamo integrare anche lo status deleted ma ne potrebbero servire anche altri .In questo momento purtroppo gli ordini vengono creati senza stato però la logica la puoi scrivere lo stesso. Per il controllo manu diceva di tornare una generica risposta di ok senza fare nulla, volendo puoi anche sovrascrivere l'orderStatus, tanto penso questo end Point sarà usato solo su id specifici già filtrati in qualche modo

Reply:
intendevo controlla semplicemente che l'ordine esista, penso puoi ignorare lo stato

Reply:
Sì ho capito cosa intendesse Ema, però nella daily del giorno dopo mi era parso di capire dopo la discussione anche con Mattia che avevamo concluso di lasciare il metodo così e di correggere solo l'eccezione e l'if-else.

Reply:
Potresti aggiungere nell input del metodo l'intero ordine, in modo da effettuare una PUT e non una PATCH, usando quindi il request body, oppure usare la patch con il request param